### ISSUE TITLE
Extending overrides in sharable configs

### ISSUE THREAD

**Issue by mrmckeb** [eslint/eslint#12685]
**The version of ESLint you are using.**
6.7.x

**The problem you want to solve.**
We have a shareable config that we use for Create React App ([`eslint-config-react-app`](https://www.npmjs.com/package/eslint-config-react-app)).

When we added TypeScript support, we added an `overrides` object for Typescript, which sets the rules for related files. JS files are still handled by the root/base config.

This works fine until users try to _extend_ these rules, they need to reimplement the TypeScript overrides object. This is causing issues for some of our users (https://github.com/facebook/create-react-app/issues/7776).

**Your take on the correct solution to problem.**
We've given it some thought. One option is for us to ship a separate config for TypeScript, and simply extend that in our overrides. That would be fairly simple for users to reimplement.

However, we wondered if the ESLint team had better ideas. Perhaps a way to merge/extend overrides?

**Are you willing to submit a pull request to implement this change?**
Of course.

**Comment by mysticatea** [comment]
Thank you for your report.

First, all content of the extender overrides the extendees at all. This is concept.

I don't understand the context completely, but I'm guessing that end users want to override settings for JS but not for others?
If it is, it's simple.

```yml
extends: original
overrides:
- files: "*.js"
  rules:
    # override some rules.
```

**Comment by mrmckeb** [comment]
Thanks for the fast response!

It's actually the other way, @mysticatea. Some want to extend the TS rules - which are implemented as an override. 

We want people to be able to add rules here, without reimplementing everything else in this override.
https://github.com/facebook/create-react-app/blob/master/packages/eslint-config-react-app/index.js#L55

**Comment by mysticatea** [comment]
Is it good with `extends`? The `extends` feature exists in order to override a part of existing configuration.

```yml
extends: original
overrides:
- files: "*.ts"
  rules:
    # override some rules.
```

**Comment by mrmckeb** [comment]
Thanks @mysticatea, you were helpful. I think the issue is on our side. I'll close this off for now.
